### ISSUE TITLE
Having Consistency+ and Universal Graves in a modpack causes multiblocks from other mods to break on a dedicated server; probaby an incompatibility with Polymer

### ISSUE THREAD

**Issue by SplendidAlakey** [ConsistencyPlus/ConsistencyPlus#108]
**Describe the bug:**

When I add Consistency+ to my modpack and join a server with the same modpack, any multiblocks from different mods stop rendering, render incorrectly or only render correctly when a block near them is updated (by either simply right-clicking or placing a block on it).

Originally, I reported the issue on the Waystones mod github here: https://github.com/ModdingForBlockheads/Waystones/issues/546, but after further testing came to a conclusion that the only reliable way to replicate the issue was to install Consistency+. 

Initially, I thought it was the problem with Waystones mod because that's the first mod which I found to be broken on the server, but after hours of testing different mod combinations I found out that just having Consistency+ in the modpack will cause multiblocks from other mods to break just like Waystones. However, not a single mod directly conflicts with Consistency+, meaning if you load just X mod and Consistency+ there won't be any issues. But as soon as you cross a certain amount of mods that add new blocks and Consistency+ is present, multiblocks break. Without Consistency+ no issues appear, even if you add a lot more mods to the pack.

**Versions:**

Minecraft version: 1.18.2
C+ version: 0.5.0 (in my logs I was using RC3 version, however, the release version also causes the issue)
Fabric API version: 0.51.0

**Other mods:**

Full list of mods required to reproduce the issue: https://gist.github.com/Footage2-Amply-Pounce/7eea0c2703631e73a3ad1a945c3b36c5 (Carpet, its addons and Fabrication can be ignored, since they don't add any new blocks)

**Logs:**

Logs following reproduction steps 1 and 2 (everything is fine): https://gist.github.com/Footage2-Amply-Pounce/ef09f9862a8d7523446a6aa8fa8362b0

Logs following reproduction steps 3 and 4 (multiblocks break): https://gist.github.com/Footage2-Amply-Pounce/a512bfe30c51b965a5ae9e14e4eccfb0

**To Reproduce:**

1. Install these mods: https://gist.github.com/Footage2-Amply-Pounce/7eea0c2703631e73a3ad1a945c3b36c5
2. Launch the game, join a server with the same mods; place a waystone, make sure it works
3. Close the game, install any of the following mods: [Adorn](https://www.curseforge.com/minecraft/mc-mods/adorn/files/3725448), [Blockus](https://www.curseforge.com/minecraft/mc-mods/blockus/files/3755903), [Chipped](https://www.curseforge.com/minecraft/mc-mods/chipped/files/3748666) or, possibly, any other mod that adds a lot of new blocks. I chose Adorn.
4. Launch the game, join the server and observe that waystones are now visually broken. However, Waystones isn't the only affected mod, if you place cables from the Tom's Simple Storage mod next to each other so they connect, you can also observe that, once connected, they become invisible until you right-click a block next to it (or place a block next to it). **edit:** Another example of broken multiblocks: beds from MultiBeds; platforms from Platforms mod; escalators, automatic doors and all the other multiblocks from Minecraft Transit Railway, openable and connectible windows from Macaw's Windows.

**Alternatively:**

1. Install all the same mods, or even way more of your choice (I believe they must be adding new blocks, not even necessarily multiblocks)
2. Make sure that everything is working as intended
3. *Then* add Consistency+ and observe that all the multiblocks broke

**Expected behavior:**

Multiblocks from any mod should render correctly at all times.

**Screenshots:**

<details>
  <summary>Broken Waystones as an example, the same applies to all multiblocks I mentioned previously</summary>

Placing a waystone down, you can see that the waystone doesn't render at all
![2022-04-09_09 47 32](https://user-images.githubusercontent.com/86702831/162562910-d0b64680-704f-4417-a715-c14172118234.png)
![2022-04-09_09 47 35](https://user-images.githubusercontent.com/86702831/162562914-b2255fe9-ad4b-452f-a4a2-0c4b31221daf.png)

Right clicking on the ground next to the waystone with an empty hand, or the waystone itself, renders the bottom half of the waystone
![2022-04-09_09 47 38](https://user-images.githubusercontent.com/86702831/162562963-f2e866c8-67a1-4570-8e30-4054798c4215.png)

Placing 2 blocks next to it (or even where it's actually supposed to be) makes the whole waystone render, but incorrectly - runes aren't glowing, no particles around the stone
![2022-04-09_09 47 53](https://user-images.githubusercontent.com/86702831/162563925-cc8ce754-12b6-4c25-a86d-0fe38c167b1c.png)

Right clicking a waystone (even when it's invisible) opens the GUI like normal, but the name is "invalid" and can't be changed
![2022-04-09_09 47 56](https://user-images.githubusercontent.com/86702831/162563027-e1c5504d-c525-44ac-849e-1c4acead074f.png)

</details>

**Additional info:**

The issue does not happen in single-player at all. Let me know if I can provide any more info.

**Comment by Cart3r1234** [comment]
So the issue is that with mods/modpacks that add large amounts of blocks, once a certain threshold is reached, multiblocks stop rendering properly until updated?

**Comment by Footage2-Amply-Pounce** [comment]
Yes, but only when Consistency+ is also installed, and only in multiplayer. Having all the same mods, and even more, but without Consistency+ doesn't produce the issue.

Also when you "fix" it by updating nearby blocks, it only stays "fixed" until you interact with it again. For instance, if you open a window from Macaw's Windows. Waystones will only properly render if you reload the chunk its in and break after you teleport, but it's the only outlier I could find, all the other multiblocks remain rendered until interacted with.

**Comment by Siuolthepic** [comment]
I can quickly cook up a test mod that outputs how many blocks were added just to see if theres an arbitrary or non-arbitrary number that has been hit.

**Comment by Footage2-Amply-Pounce** [comment]
> I can quickly cook up a test mod that outputs how many blocks were added just to see if theres an arbitrary or non-arbitrary number that has been hit.

That should at the very least rule out the amount of blocks as the reason for the bug, because right now it's really only my assumption after hours and hours of testing all sorts of mod combinations. I use a lot more mods than needed to replicate the issue, but from my testing it seems that only adding mods that introduce new blocks matters. Which is how I managed to narrow down the list and find a way to consistently reproduce the issue.

**Comment by Siuolthepic** [comment]
Try this mod (Unzip it before adding)
[projectdiag-1.0.0.zip](https://github.com/ConsistencyPlus/ConsistencyPlus/files/8548479/projectdiag-1.0.0.zip)

**Comment by Siuolthepic** [comment]
Also, note that the mod is intended to get that block count

**Comment by Footage2-Amply-Pounce** [comment]
> Try this mod and send log of it (Unzip it before adding) [projectdiag-1.0.0.zip](https://github.com/ConsistencyPlus/ConsistencyPlus/files/8548479/projectdiag-1.0.0.zip)

Sorry, took a while, followed my steps to reproduce the issue + added the test mod. Also for some reason just adding one extra mod during step 3 is no longer enough, so I added all 2 instead (Adorn + Blockus). Interestingly, it doesn't break waystones anymore, unless I install my entire modpack, but openable windows from Macaw's Windows, MultiBeds and all the other multiblocks I tested still break with just the selection of mods I listed before.

Here are the results: https://gist.github.com/Footage2-Amply-Pounce/917c1095f7e03c9d373fb74fc62e52d4

If you want me to test anything further, I'll have to do it tomorrow, but I will try to help as much as I can.

**Comment by Siuolthepic** [comment]
Blockcount was at 7779.

Closest possible arbitrary and code limits I see near it are 4096, 5000, and 7500.
Test with as little mods as possible if you can.

**Comment by Footage2-Amply-Pounce** [comment]
I reduced the amount of mods needed to basically the essentials: https://gist.github.com/Footage2-Amply-Pounce/7eea0c2703631e73a3ad1a945c3b36c5.

However, with these mods, not *all* multiblocks break. First of all, Waystones no longer break as I mentioned yesterday, unless way more mods are installed, as such I excluded it from this list and tests. Second, Macaw's Windows don't break with this selection of mods, to reproduce the issue with windows you need a couple more, I added [Decorative Blocks](https://www.curseforge.com/minecraft/mc-mods/decorative-blocks/files/3681918) and its addon [Decorative Winter](https://www.curseforge.com/minecraft/mc-mods/decorative-winter/files/3681948). Third, MultiBeds' beds aren't broken with these mods, you need to add 1 more for it to break, I added [Decorative Blocks](https://www.curseforge.com/minecraft/mc-mods/decorative-blocks/files/3681918) without the addon. Multiblocks from Platforms still consistently get broken.

I don't think the block count has anything to do with it (though, just for the record, with this limited selection of mods it reaches 5767), but I still believe C+ does, because removing it solves the issue, even if I add a lot more mods to the pack.

Something of note I found out while testing this selection of mods: removing C+ solves the issue; removing Universal Graves solves the issue; removing Adorn and its dependency Kotlin solves the issue; removing Simple copper pipes solves the issue. Removing anything other than C+ from a large modpack does not solve the issue, it only does with this very limited selection of mods and only if C+ is also installed.

**Comment by Footage2-Amply-Pounce** [comment]
Out of curiosity launched my entire mod pack with C+ and without, with C+ the Block count is at 11598 and all sorts of multiblocks are broken, without - 8969 and not a single multiblock is broken, so that 7779 block count was probably not the issue.

Is it possible that C+ somehow prevents the server from properly sending packets to the client or something like that? Because the server is aware that there is a block, it won't let you walk through it, you will keep rubberbanding back until you force the block to appear. Very similarly to so called "ghost blocks". I have no other ideas as to what's going on here.

**Comment by Siuolthepic** [comment]
It seems unlikely, but did we somehow 5 way dependency error with these?

**Comment by Footage2-Amply-Pounce** [comment]
> It seems unlikely, but did we somehow 3-5 way error with these?

What does "3-5 way error" mean?

**Comment by Footage2-Amply-Pounce** [comment]
[Jello](https://www.curseforge.com/minecraft/mc-mods/jello/files/3770006) causes the same issue as C+. They have C+ listed as an optional dependency, so whatever they include from C+ into Jello causes the same issue as C+ itself. Might be helpful, might not, just letting you know.

**Comment by itsdinkd** [comment]
> Out of curiosity launched my entire mod pack with C+ and without, with C+ the Block count is at 11598 and all sorts of multiblocks are broken, without - 8969 and not a single multiblock is broken, so that 7779 block count was probably not the issue. Or rather it might only become problematic if C+ is installed.
> 
> Is it possible that C+ somehow prevents the server from properly sending packets to the client or something like that? Because the server is aware that there is a block, it won't let you walk through it, you will keep rubberbanding back until you force the block to appear. Very similarly to so called "ghost blocks". I have no other ideas as to what's going on here.



yes. this happened in my modpack before when there were too many blocks (in this case, it was due to chipped). Weird stuff happened, like recipes not even working with REI. I believe there is a vanilla cap with the total amount of blocks you can have. this happened a while ago though so I forgot the outcome of the fix, I wish I could be more help. but here to say that stuff definitely happens when a threshold is hit

**Comment by Footage2-Amply-Pounce** [comment]
> yes. this happened in my modpack before when there were too many blocks (in this case, it was due to chipped). Weird stuff happened, it interfered with KubeJS and REI. I believe there is a vanilla cap with the total amount of blocks you can have. this happened a while ago though so I forgot the outcome of the fix, I wish I could be more help. but here to say that stuff definitely happens when a threshold is hit

I had a modpack on 1.17.1 with practically every fabric mod out there and had no such issues. And right now I'm running a large modpack on 1.18.2 that works completely fine as long as C+ isn't installed. I doubt it's a vanilla limit or a limit at all, that was just my guess before I ran the tests.

**Comment by itsdinkd** [comment]
My issue was on 1.18. I have 330 mods in my pack along with c+ right now
and multiblocks are working fine for me. My pack has about 10 blocks.

**Comment by itsdinkd** [comment]
10k* this is on 1.18.2

**Comment by Footage2-Amply-Pounce** [comment]
> I have 330 mods in my pack along with c+ right now
and multiblocks are working fine for me. My pack has about 10 blocks.

My issue is reproducible with just 14 mods. Maybe I should change the title.

I'm curious though about your modpack, could you post your modlist whenever you can, please? I'll compare it to mine and see if I still get the issue with your pack.

**Comment by Cart3r1234** [comment]
> My issue is reproducible with just 14 mods. Maybe I should change the title
This is certainly one of the more bizarre issues we've had reported.  Is there any chance it could be an incompatibility with a specific mod rather than of some sort of limitation that C+ hits?

**Comment by itsdinkd** [comment]
> > I have 330 mods in my pack along with c+ right now
> > and multiblocks are working fine for me. My pack has about 10 blocks.
> 
> My issue is reproducible with just 14 mods. Maybe I should change the title.
> 
> I'm curious though about your modpack, could you post your modlist whenever you can, please? I'll compare it to mine and see if I still get the issue with your pack.
> 
> Edit: also are you playing on a dedicated server? If not, then you were having a totally separate problem, as my issue happens only on a server. Single-player is unaffected.

My problem that I wrote about is irrelevant to what you mentioned about multi blocks, but yes at that time it was on both server and single player. 

```java
		additionaladditions: Additional Additions 4.0.0
		additionalbars: Additional Bars 3.0.0
		additionalentityattributes: Additionalentityattributes 1.0.0+1.18.2
		adorn: Adorn 3.4.0+1.18.2
		advanced_runtime_resource_pack: Runtime Resource Pack 0.5.7
		advdebug: Advancements Debug 2.3.0
		adventurez: AdventureZ 1.4.10
		ae2: Applied Energistics 2 11.0.0-alpha.5
		ae2wtlib: AE2WTLib 11.0.0-alpha.2
		agape_space: Space Dimensions 0.5.8
		alloy_forgery: Alloy Forgery 2.0.13+1.18
		annotateddi: Annotated DI 2.1.0+1.18.2
		aopalliance_aopalliance: aopalliance 1.0
		apoli: Apoli 2.3.3
		appleskin: AppleSkin 2.4.0+mc1.18.2
		arcanus: Arcanus 1.29
		architects_palette: Architect's Palette Fabric 1.7
		architectury: Architectury 4.2.50
		armorpointspp: Armorpoints++ 3.0.0
		artis: Artis-Recrafted 2.4.0+1.18.2
		attributefix: AttributeFix 14.0.1
		autoconfig1u: Auto Config v1 Updated 3.4.0
		autorun: AutoRun 0.4.0
		badpackets: Bad Packets 0.1.2
		balm-fabric: Balm 3.0.0+0
		banhammer: BanHammer 0.5.0+1.18.2
		bcc: BetterCompatibilityChecker fabric-1.0.5
		bclib: BCLib 1.4.5
		beenfo: Beenfo 1.18.2-fabric0.47.10-1.3.3
		betteradvancements: Better Advancements 0.2.0.129
		betteranimalsplus: Better Animals Plus 1.18.2-11.0.5
		betterend: Better End 1.1.0
		betternether: Better Nether 6.1.1
		bewitchment: Bewitchment 1.18-17
		bhmenu: BHMenu-Fabric 1.17.1-1.18.2-1.3
		bitsandchisels: BitsAndChisels 2.6.8
		blockbot-api: BlockBot Api 1.0.0
		blockbot-discord: BlockBot Discord 2.0.0-beta
		blockus: Blockus 2.4.5+1.18.2
		blue_endless_jankson: jankson 1.2.1
		bno: Basic Nether Ores 1.18.2-7.2.0-Fabric
		bookshelf: Bookshelf 13.0.13
		bosses_of_mass_destruction: Bosses of Mass Destruction (Beta) 1.3.6-1.18.2
		botania: Botania 1.18.2-431-FABRIC
		buildinggadgets: st'ructure tools 1.0.21
		byg: Oh The Biomes You'll Go 1.4-RC-10
		cacao: Croptopia's Chocolaterie 1.1
		calio: Calio 1.5.1
		capybara: Capybara 1.0.4
		cardinal-components: Cardinal Components API 4.1.4
		cardinal-components-base: Cardinal Components API (base) 4.1.4
		cardinal-components-block: Cardinal Components API (blocks) 4.1.4
		cardinal-components-chunk: Cardinal Components API (chunks) 4.1.4
		cardinal-components-entity: Cardinal Components API (entities) 4.1.4
		cardinal-components-item: Cardinal Components API (items) 4.1.4
		cardinal-components-level: Cardinal Components API (world saves) 4.1.4
		cardinal-components-scoreboard: Cardinal Components API (scoreboard) 4.1.4
		cardinal-components-world: Cardinal Components API (worlds) 4.1.4
		carpet: Carpet Mod 1.4.69
		carpet-extra: Carpet Extra 1.4.72
		carpet-tis-addition: Carpet TIS Addition 1.33.0
		carrier: Carrier 1.10.0
		catwalksinc: Catwalks Inc. 1.3.0
		chalk: Chalk 2.0.0+1.18.2
		charm: Charm 4.1.0
		chisel: Chisel Refabricated 1.4.5
		clienttweaks: Client Tweaks 7.1.0
		cloth-api: Cloth API 3.2.64
		cloth-basic-math: cloth-basic-math 0.6.0
		cloth-client-events-v0: Cloth Client Events v0 3.2.64
		cloth-common-events-v1: Cloth Common Events v1 3.2.64
		cloth-config: Cloth Config v6 6.2.57
		cloth-config-lite: cloth-config-lite 2.0.6
		cloth-datagen-api-v1: Cloth Datagen v1 3.2.64
		cloth-utils-v1: Cloth Utils v1 3.2.64
		clumps: Clumps 8.0.0+8
		com_electronwill_night-config_core: core 3.6.4
		com_electronwill_night-config_toml: toml 3.6.4
		com_eliotlash_mclib_mclib: mclib 18
		com_eliotlash_molang_molang: molang 18
		com_fasterxml_jackson_core_jackson-annotations: jackson-annotations 2.13.1
		com_fasterxml_jackson_core_jackson-core: jackson-core 2.13.1
		com_fasterxml_jackson_core_jackson-databind: jackson-databind 2.13.1
		com_flowpowered_flow-math: flow-math 1.0.3
		com_github_dimensionaldevelopment_poly2tri_java: poly2tri.java 0.1.1
		com_google_inject_guice: guice 5.1.0
		com_moandjiezana_toml_toml4j: toml4j 0.7.2
		com_typesafe_config: config 1.4.1
		com_velocitypowered_velocity-native: velocity-native 1.1.0-SNAPSHOT
		computercraft: CC: Restitched 1.100.5
		conditional-mixin: conditional mixin 0.2.0
		config2brigadier: Config to Brigadier 1.1.1
		conjuring: Conjuring 1.0.22+1.18
		consistency_plus: Consistency Plus 0.5.0+1.18.2
		cosmetic-armor: Cosmetic Armor 1.3.1
		craftingtweaks: Crafting Tweaks 13.0.0
		crawl: Crawl 0.10.6
		creativecore: CreativeCore (Fabric) 2.6.10
		creeperspores: Creeper Spores 1.8.0
		croparia: Croparia 3.8.3
		croptopia: Croptopia 2.0.0
		crowdin-translate: CrowdinTranslate 1.4+1.18.2
		curse_maven_wthit-3707010: wthit-3707010 440979
		customportalapi: Custom Portal Api 0.0.1-beta50-1.18
		damagetilt: Damage Tilt 1.18-fabric-0.1.2
		dankstorage: Dank Storage 1.18.2-3.7
		dark-enchanting: Dark Enchanting 0.7.0+1.18.2
		darkutils: DarkUtilities 10.0.4
		dawn: Dawn API 3.3.0
		de_poiu_apron_apron: apron 2.1.1
		deathlog: DeathLog 0.2.7+1.18
		dimdoors: DimensionalDoors 4.0.4
		disenchanter: Disenchanter 0.1.6+1.18
		dml-refabricated: Deep Mob Learning: Refabricated 1.18.2
		dml-refabricated-base: DML: Refabricated - Base 1.18.2
		dml-refabricated-events: DML: Refabricated - Vanilla Events 1.18.2
		dml-refabricated-modular-armor: DML: Refabricated - Glitch Armor 1.18.2
		dml-refabricated-rei: DML: Refabricated - REI Integration 1.18.2
		dmlsimulacrum: Deep Mob Learning: Simulacrum 0.5.3-beta
		doom: Doom Fabric 4.2.3
		doublejump: Double Jump Mod 1.0.1+mc1.18
		dragonloot: DragonLoot 1.0.10
		dripstone_fluid_lib: Dripstone Fluid Lib 1.2.0
		dummmmmmy: MmmMmmMmmMmm 1.18-1.5.0
		dusk: Dragonfight Mod 1.18.2-2.1
		earthbounds: Earthbounds 1.18.2-1.1.0b
		earthtojavamobs: Earth2Java 1.8.0+1.18.2
		eldritch_mobs: Eldritch Mobs 1.11.0
		elementalcreepers: Elemental Creepers: Refabricated 1.4.2+1.18.2
		enchantment_force: enchantment_force 0.1.7
		exlineawnings: Exline's Awnings fabric-1.18.2-v1.0.2
		exlinefurniture: Exline's Furniture fabric-1.18.2-v2.0.5
		expandedstorage: Expanded Storage 7.3.12
		experiencebugfix: Experience Bug Fix 18
		explorerscompass: Explorer's Compass 1.18.2-2.1.0-fabric
		extended_drawers: Extended Drawers 1.0.2+mc.1.18.2
		extragenerators: Extra Generators 1.2.1-BETA+1.18
		extraorigins: Extra Origins 1.18-10
		fabric: Fabric API 0.51.1+1.18.2
		fabric-api-base: Fabric API Base 0.4.4+d7c144a860
		fabric-api-lookup-api-v1: Fabric API Lookup API (v1) 1.6.0+2373a54560
		fabric-biome-api-v1: Fabric Biome API (v1) 8.0.4+7b0f312660
		fabric-command-api-v1: Fabric Command API (v1) 1.1.9+d7c144a860
		fabric-commands-v0: Fabric Commands (v0) 0.2.8+b4f4f6cd60
		fabric-containers-v0: Fabric Containers (v0) 0.1.21+d7c144a860
		fabric-content-registries-v0: Fabric Content Registries (v0) 3.0.3+d82b939260
		fabric-convention-tags-v1: Fabric Convention Tags 1.0.1+474e4f7960
		fabric-crash-report-info-v1: Fabric Crash Report Info (v1) 0.1.11+3ac43d9560
		fabric-diagonal-panes: Fabric Diagonal Panes 0.6.1
		fabric-dimensions-v1: Fabric Dimensions API (v1) 2.1.17+2540745460
		fabric-entity-events-v1: Fabric Entity Events (v1) 1.4.8+d7c144a860
		fabric-events-interaction-v0: Fabric Events Interaction (v0) 0.4.19+d7c144a860
		fabric-events-lifecycle-v0: Fabric Events Lifecycle (v0) 0.2.14+d7c144a860
		fabric-furnaces: Fabric Furnaces 2.1.1-1.18.2
		fabric-game-rule-api-v1: Fabric Game Rule API (v1) 1.0.13+d7c144a860
		fabric-item-api-v1: Fabric Item API (v1) 1.4.0+4457765560
		fabric-item-groups-v0: Fabric Item Groups (v0) 0.3.11+91896a4960
		fabric-language-kotlin: Fabric Language Kotlin 1.7.4+kotlin.1.6.21
		fabric-language-scala: Fabric Language Scala 1.1.0+scala.2.13.6
		fabric-lifecycle-events-v1: Fabric Lifecycle Events (v1) 2.0.3+2d83e92c60
		fabric-loot-tables-v1: Fabric Loot Tables (v1) 1.0.11+d7c144a860
		fabric-mining-level-api-v1: Fabric Mining Level API (v1) 2.1.0+81ea5b3c60
		fabric-networking-api-v1: Fabric Networking API (v1) 1.0.21+d882b91560
		fabric-networking-v0: Fabric Networking (v0) 0.3.8+b4f4f6cd60
		fabric-object-builder-api-v1: Fabric Object Builder API (v1) 2.1.0+fe4ddef060
		fabric-particles-v1: Fabric Particles (v1) 0.2.12+526dc1ac60
		fabric-permissions-api-v0: fabric-permissions-api 0.1-SNAPSHOT
		fabric-registry-sync-v0: Fabric Registry Sync (v0) 0.9.8+0d9ab37260
		fabric-rendering-data-attachment-v1: Fabric Rendering Data Attachment (v1) 0.3.7+d7c144a860
		fabric-rendering-fluids-v1: Fabric Rendering Fluids (v1) 2.0.2+54e5b2ec60
		fabric-resource-conditions-api-v1: Fabric Resource Conditions API (v1) 2.0.3+91b7aa6660
		fabric-resource-loader-v0: Fabric Resource Loader (v0) 0.4.18+2de5574560
		fabric-screen-handler-api-v1: Fabric Screen Handler API (v1) 1.2.0+1f6558e860
		fabric-transfer-api-v1: Fabric Transfer API (v1) 1.6.0+2373a54560
		fabric-transitive-access-wideners-v1: Fabric Transitive Access Wideners (v1) 1.0.0+bd8a4a1860
		fabricenchantments: Fabric Enchantments 0.7.2
		fabricloader: Fabric Loader 0.13.3
		fabricshieldlib: Fabric Shield Lib 1.5.3-1.18
		fabrictailor: Fabric Tailor 1.8.1
		fallingtree: FallingTree 3.5.0
		farmersdelight: Farmer's Delight 1.18.2-0.2.1
		farmingforblockheads: Farming for Blockheads 10.0.0
		feature_nbt_deadlock_be_gone: Feature NBT Deadlock Be Gone - Fabric 2.0.0+1.18.2-rc1
		ferritecore: FerriteCore 4.2.1
		fiber: fiber 0.23.0-2
		fireplacelib: Fireplace Lib 6.3.1+1.18.2
		fireworkfrenzy: Firework Frenzy 1.9
		flan: Flan 1.18.2-1.7.6
		fluidtank: Fluid Tank 18.5.0-fabric
		forgeconfigapiport: Forge Config API Port 3.2.0
		ftblibrary: FTB Library 1802.3.6-build.115
		ftbquests: FTB Quests 1802.3.8-build.101
		ftbteams: FTB Teams 1802.2.6-build.51
		gearreborn: Gear Reborn 1.18.2-0.1.12
		geckolib3: Geckolib 3.0.38
		genesis: Genesis 1.18.2-1.0.1
		givemehats: Give Me Hats! 1.8.0
		go-fish: Go Fish 1.5.0-1.18.2
		gobber2: Gobber2 2.5.72
		goldenhoppers: Golden Hoppers 1.3.0
		graveyard: The Graveyard 1.7
		graveyard_biomes: The Graveyard Biomes 1.1
		grounded_origins: Grounded Origins 1.1.3
		guardiansgalore: GuardiansGalore 1.18.2-3.0
		gud_recipe_confidence: Recipe Confidence 1.0.2
		hammermod: War-Hammers Mod fabric-1.18.2-v1.0.1
		harvest_scythes: Harvest Scythes 2.5.0
		hookshot: Hookshot 25.0
		horseinfo: Horse Info 0.4.1-1.18
		horsestatsvanilla: Horse Stats Vanilla 4.1.10
		hoverpets: Hover Pets 0.3.2
		hyperlighting: Hyper Lighting 1.2.2
		iceberg: Iceberg 1.0.38
		impaled: Impaled 1.1.1
		impersonate: Impersonate 2.5.0
		improved-stations: Improved Stations 2.1.0
		incantationem: Incantationem 1.1.3+1.18.2
		incubus_core: Incubus Core 1.6.6+1.18.2
		inmis: Inmis 2.5.3-1.18.2
		interference: Interference API 1.3.9
		io_leangen_geantyref_geantyref: geantyref 1.3.11
		itemfilters: Item Filters 1802.2.7-build.25
		jankson: Jankson 4.0.0+j1.2.0
		java: OpenJDK 64-Bit Server VM 17
		javax_inject_javax_inject: javax.inject 1
		jsonem: JsonEM 0.1.1
		justenoughkeys: Just Enough Keys 0.5.3
		kanos_config: Kanos Config 0.2.0+1.14.4-1.19
		kibe: Kibe 1.9.7-BETA+1.18
		kjsextras: KubeJS Extras 2.0.0
		konkrete: Konkrete 1.3.3
		krypton: Krypton 0.1.9
		ktdatataglib: KtDataTagLib 1.6.5+1.18.2
		kubejs: KubeJS 1802.5.1-build.372
		kyrptconfig: Kyrpt Config 1.3.4-1.18
		lazydfu: LazyDFU 0.1.2
		libblockattributes: LibBlockAttributes 0.10.2
		libblockattributes_core: LibBlockAttributes (Core) 0.10.2
		libblockattributes_fluids: LibBlockAttributes (Fluids) 0.10.2
		libblockattributes_items: LibBlockAttributes (Items) 0.10.2
		libgui: LibGui 5.3.2+1.18.2
		libninepatch: LibNinePatch 1.1.0
		libra: Libra 0.4.2
		lifts: Lifts 1.4.1-BETA+1.18
		lightestlamp: Lightest Lamps 1.0.0
		lithium: Lithium 0.7.9
		lovely_snails: Lovely Snails 1.0.4+1.18
		luckperms: LuckPerms 5.4.21
		macawsbridgesbyg: Macaw's Bridges - BYG 1.18.2-1.0
		maelstrom_library: Maelstrom Library 1.2.4-1.18.2
		magiccombatwands: Magic Combat Wands 1.0.1
		magicfungi: Magic Fungi 0.7.1-BETA+1.18.2
		magna: Magna 1.7.0-1.18-pre1
		marketcrates: Marketcrates 1.4.1
		matrix: Matrix 1.0.0
		mavm: More Axolotl Variants Mod 1.0.7a
		maxhealthfix: MaxHealthFix 5.0.1
		mcda: MC Dungeons Armors 2.0.6
		mcdar: MC Dungeons Artifacts 1.4.7
		mcdw: MC Dungeons Weapons 4.3.2
		mconfig: Mconfig 1.1.2
		mcwbridges: Macaw's Bridges 2.0.2
		mcwfences: Macaw's Fences and Walls 1.0.4
		mcwlights: Macaw's Lights and Lamps 1.0.3
		mcwpaintings: Macaw's Paintings 1.0.3
		mcwwindows: Macaw's Windows 2.0.3
		me_hypherionmc_night-config_core: core 3.6.5_custom
		me_hypherionmc_night-config_toml: toml 3.6.5_custom
		megane: megane 7.5.1
		megane-api: megane-api 7.1.0
		megane-applied-energistics-2: megane-applied-energistics-2 7.1.0
		megane-deep-mob-learning-simulacrum: megane-deep-mob-learning-simulacrum 7.1.0
		megane-extra-generators: megane-extra-generators 7.1.0
		megane-fabric-transfer: megane-fabric-transfer 7.1.0
		megane-kibe: megane-kibe 7.1.0
		megane-lib-block-attributes: megane-lib-block-attributes 7.1.0
		megane-reborn-core: megane-reborn-core 7.1.0
		megane-reborn-energy: megane-reborn-energy 7.0.0
		megane-runtime: megane-runtime 7.2.0
		megane-tech-reborn: megane-tech-reborn 7.1.0
		megane-vanilla: megane-vanilla 7.1.1
		megane-wireless-networks: megane-wireless-networks 7.1.0
		midnightlib: MidnightLib 0.4.0
		milk: Milk Lib 0.3.1
		milk_plus: Milk+ 3.0.1
		minecraft: Minecraft 1.18.2
		mining_dims: Mining Dimensions 1.3.6
		miningutility: Mining Utility 1.2.1+1.18
		mm: Manningham Mills 2.3
		mo_glass: Mo Glass 1.6.1-MC1.18.2
		mobscarecrow: Mob Scarecrow 1.2.2
		mobz: MobZ 3.3.1
		mocolors: Mo' Colors 1.4.0
		modern_industrialization: Modern Industrialization 1.1.0-alpha.01
		morebannerfeatures: More Banner Features 1.1.3
		moretags: More Tags 3.0.3
		morevillagers-fabric: MoreVillagersFabric 3.2.0-SNAPSHOT
		mostructures: Mo' Structures 1.4.1+1.18.2
		mtmechs: Magitek Mechs 1.0.12
		multibeds: MultiBeds 1.2.5
		multipart_entities: MultipartEntities 1.1.5-1.18.2
		myloot: myLoot 2.1
		myron: Myron 1.6.3+1.18.1
		mysql_mysql-connector-java: mysql-connector-java 8.0.28
		mythicmetals: Mythic Metals 0.13.1
		mythicmounts: Mythic Mounts 1.18.2-7.0
		naturescompass: Nature's Compass 1.18.2-2.0.3-fabric
		net_objecthunter_exp4j: exp4j 0.4.8
		ninjaphenix_container_lib: NinjaPhenix's Container Library 1.3.0
		notenoughcrashes: Not Enough Crashes 4.1.6+1.18.2
		omega-config: OmegaConfig 1.2.3-1.18.1
		omega-config-gui: Omega Config GUI 1.2.3-1.18.1
		onsoulfire: On Soul Fire 1.18-1
		org_jetbrains_kotlin_kotlin-reflect: kotlin-reflect 1.6.21
		org_jetbrains_kotlin_kotlin-stdlib: kotlin-stdlib 1.6.21
		org_jetbrains_kotlin_kotlin-stdlib-jdk7: kotlin-stdlib-jdk7 1.6.21
		org_jetbrains_kotlin_kotlin-stdlib-jdk8: kotlin-stdlib-jdk8 1.6.21
		org_jetbrains_kotlinx_kotlinx-coroutines-core-jvm: kotlinx-coroutines-core-jvm 1.6.1
		org_jetbrains_kotlinx_kotlinx-coroutines-jdk8: kotlinx-coroutines-jdk8 1.6.1
		org_jetbrains_kotlinx_kotlinx-serialization-cbor-jvm: kotlinx-serialization-cbor-jvm 1.3.2
		org_jetbrains_kotlinx_kotlinx-serialization-core-jvm: kotlinx-serialization-core-jvm 1.3.2
		org_jetbrains_kotlinx_kotlinx-serialization-json-jvm: kotlinx-serialization-json-jvm 1.3.2
		org_jgrapht_jgrapht-core: jgrapht-core 1.1.0
		org_spongepowered_configurate-core: configurate-core 4.1.2
		org_spongepowered_configurate-hocon: configurate-hocon 4.1.2
		org_xerial_sqlite-jdbc: sqlite-jdbc 3.36.0.3
		org_yaml_snakeyaml: snakeyaml 1.30
		origins: Origins 1.4.1
		origins-classes: Origins: Classes 1.3.0
		overlord: Overlord 3.1.5+1.18.2
		overweight_farming: Overweight Farming 1.18.2-1.2.0-fabric
		owo: oωo 0.7.0+1.18
		packet_tweaker: Packet Tweaker 0.2.0+1.18-pre1
		paintings: Paintings++ 9.1.1.0
		patchouli: Patchouli 1.18.2-67-FABRIC
		pehkui: Pehkui 3.2.2+1.14.4-1.19
		phonos: Phonos 0.3+1.18.2
		phosphor: Phosphor 0.8.1
		pickupnotifier: Pick Up Notifier 3.2.0
		pigpen: PigPen 8.0.1
		placeholder-api: Placeholder API 1.1.3+1.17.1
		platforms: Platforms 1.2.4
		player-data-api: Player Data API 0.1.0+1.18
		playerabilitylib: Pal 1.5.1
		plushies: Plushie Mod 1.2
		polymorph: Polymorph 0.0.20-1.18.2
		prefab: Prefab 3.1.3
		publicguiannouncement: Public Gui Announcement 3.2.0.1
		pugh_lib: Pugh Lib 1.3.0
		pugh_tools: Pugh Tools 1.2.5
		puzzleslib: Puzzles Lib 3.3.1
		quarryplus: QuarryPlus 18.4.555
		quartzelv: Quartz Elevator 2.1.1+1.18
		quickshulker: Quick Shulker 1.3.6-1.18
		reach-entity-attributes: Reach Entity Attributes 2.1.1
		reborncore: Reborn Core 5.2.0-beta.3
		redbits: Red Bits 1.13.2
		redstonebits: Redstone Bits 1.7.1
		repurposed_structures: Repurposed Structures 4.1.6+1.18.2
		resourceful_tools: Resourceful Tools 1.2.35
		respawnablepets: Respawnable Pets 1.18-4
		rhino: Rhino 1801.1.9-build.135
		right-click-harvest: Right-Click-Harvest 1.2.0
		rocks: This Rocks! 1.6.0
		roughlyenoughitems: Roughly Enough Items 8.1.457
		roughlyenoughresources: Roughly Enough Resources 2.5.0
		rpgstats: RPGStats 4.3.8+1.18.2
		runelic: Runelic 11.0.1
		sandwichable: Sandwichable 1.2+1.18.2
		secretrooms: Secret Rooms 1.3.1+1.18.2
		seedfix: Seed Fix 1.0.1
		server_translations_api: Server Translations API 1.4.10+1.18.2
		sgui: sgui 1.0.1+1.18.2
		shetiphiancore: ShetiPhianCore 1.2.10
		shulkerboxtooltip: Shulker Box Tooltip 3.0.7+1.18.2
		shulkerutils: shulkerutils 1.0.4-1.18
		silkspawners: SilkSpawners 1.1.1
		someforgepatchesported: Some Forge Patches Ported 1.0.3
		spark: spark 1.9.1
		spectrum: Spectrum 1.3.3-1.18.2-midgame18-trinkets-brewing
		spoornpacks: SpoornPacks 4.2.1-1.18.2
		spoornweaponattributes: SpoornWeaponAttributes 3.4
		stacc: Stacc 1.2.3
		step-height-entity-attribute: Step Height Entity Attribute 1.0.0
		stonecutter_recipe_tags: Stonecutter Recipe Tags 3.0.1+1.18.bc850d5.fabric
		stoneholm: Stoneholm 1.4.3
		styledchat: Styled Chat 1.2.3+1.18
		styledplayerlist: Styled Player List 2.1.3+1.18
		tacocraft: TacoCraft 1.7.4-1.18.2
		team_reborn_energy: Energy 2.2.0
		techreborn: Tech Reborn 5.2.0-beta.3
		teenycoal: Teeny Coal 1.1.2-1.16.x-1.17.x-1.18.x-1.19-experimental-snapshot
		tempad: Tempad 1.1.0
		tenshilib: TenshiLib 1.18.2-1.6.7
		terrablender: TerraBlender 1.1.0.99
		terraform-wood-api-v1: Terraform Wood API (v1) 3.0.0
		terrarianslimes: Terrarian Slimes 1.2-BETA+1.18
		the_aether: Paradise Lost 1.6.8+1.18.2
		things: Things 0.2.17+1.18
		tinyskeletons: Tiny Skeletons 3.2.1
		toms_storage: Tom's Simple Storage Mod 1.3.0
		toolstats: ToolStats 9.0.1
		tradingpost: Trading Post 3.2.0
		trashslot: TrashSlot 11.0.2
		trinkets: Trinkets 3.3.0
		tweed4_annotated: tweed4_annotated 1.3.0
		tweed4_base: tweed4_base 1.5.0+mc1.18.2
		tweed4_data: tweed4_data 1.2.1+mc1.18.2
		tweed4_data_hjson: tweed4_data_hjson 1.1.0
		twigs: Twigs 1.1.3-patch1+1.18.2
		unforgivingvoid: Unforgiving Void 3.1.4+1.18.2
		valley: ValleyCraft 2.3-1.18.2
		veggie_way: The Veggie Way 1.2.32
		veinmining: Vein Mining 0.0.12-1.18.2
		villager-hats: Villager Hats Mod 1.3.1
		villagernames: Villager Names 4.0.4
		visualworkbench: Visual Workbench 3.3.0
		voidz: VoidZ 1.0.7
		wandering_collector: Wandering Collector 1.1.2+mc1.18.2
		waterdripsound: Drip Sounds 1.18-0.3.0
		waystones: Waystones 2.5.0
		whisperwoods: Whisperwoods 1.18.2-2.1.0
		winged: Winged Mod 1.18.2-3.2.0
		wirelessnetworks: Wireless Networks 1.5.1
		wthit: wthit 4.9.1
		xaerominimap: Xaero's Minimap 22.6.1
		xaeroworldmap: Xaero's World Map 1.20.7
		xlpackets: XLPackets 2.1-1.17.1
		xps: Xp Obelisk 0.3.8.r_for_1.18.x_Fabric
		yigd: You're in Grave Danger 1.3.1
		zihlaunchpads: ZIHLaunchpads 1.2.0
		```

**Comment by Footage2-Amply-Pounce** [comment]
> This is certainly one of the more bizarre issues we've had reported. Is there any chance it could be an incompatibility with a specific mod rather than of some sort of limitation that C+ hits?

I tested all the mods required to reproduce the issue 1 by 1 with C+ and none cause the same problem as all of them together.

And yeah, it's one of those problems where things don't really make sense to me too.

**Comment by Siuolthepic** [comment]
This is so strange.

My theory is currently a block state or block model limit.
It would explain how some block counts don't trigger this but others do

**Comment by Cart3r1234** [comment]
Also it having to do with multiplayer may be a clue into what causes this - just as a test, does it do the same thing if _before_ you join any multiplayer worlds, you join a singleplayer world first?  We had a different issue where doing that caused different results, and at this point this is just a shot in the dark to see if anything new happens.

**Comment by Footage2-Amply-Pounce** [comment]
> My problem that I wrote about is irrelevant to what you mentioned about multi blocks, but yes at that time it was on both server and single player. [...]

Thanks for the modlist. We have a lot of mods in common, so I'm surprised you don't have the same issue... I'll try to replicate it with your mods later, but for now you are absolutely sure that with your mods, on a dedicated server, MultiBeds render correctly when placed down, Platforms render correctly when floors are placed on top of each other etc, Waystones render correctly at all times and Macaw's Windows don't disappear when being opened?

> My theory is currently a block state or block model limit.
It would explain how some block counts don't trigger this but others do

Yep, I thought about it too, C+ would be the only mod in my modpack that adds massive amounts of new shapes for existing blocks. All the other ones simply add new ones entirely. However, I don't know how would I go about testing it...

> Also it having to do with multiplayer may be a clue into what causes this - just as a test, does it do the same thing if before you join any multiplayer worlds, you join a singleplayer world first? We had a different issue with stonecutters/SCRT tags breaking where doing that caused different results, and at this point this is just a shot in the dark to see if anything new happens.

I thought of the same thing because your mod, as you mentioned, Chipped and Traveler's Backpack used to have an issue where you had to join SP first in order for things to work properly in MP, so I tried doing that and nope, loading into SP first doesn't change anything.

**Comment by Siuolthepic** [comment]
I could probably make a block that has like 2^16 blockstates and we can go from there

**Comment by TropheusJ** [comment]
Not sure this is related to block count. I think this might be some kind of EXTREMELY niche load order issue. I've seen several cases of mod counts causing issues instead of mod contents, and that seems somewhat similar here. 
> I reduced the amount of mods needed to basically the essentials: https://gist.github.com/Footage2-Amply-Pounce/7eea0c2703631e73a3ad1a945c3b36c5.

Is this still the smallest combo found?

**Comment by Footage2-Amply-Pounce** [comment]
> I could probably make a block that has like 2^16 blockstates and we can go from there

If you won't be able to test it yourself, drop the test mod here and I'll get to it within the next couple of days, can't do it today and, likely, tomorrow either.

Also I read through the entire list itsdinkd provided, seems like the only 2 mods he doesn't have, that I used to replicate the issue (https://gist.github.com/Footage2-Amply-Pounce/7eea0c2703631e73a3ad1a945c3b36c5) are Universal Graves and Simple Copper Pipes. Copper pipes have a lot of new blocks, but unlikely to be the cause of the issue. However, Universal Graves uses [Polymer](https://www.curseforge.com/minecraft/mc-mods/polymer), which now that I think of it, could be what's causing it, I don't think I tested UG and C+ alone because UG doesn't add any new blocks, but I completely overlooked the fact it uses Polymer. I'll look into that right now as that's going to be quick.

**Comment by Footage2-Amply-Pounce** [comment]
> > I reduced the amount of mods needed to basically the essentials: https://gist.github.com/Footage2-Amply-Pounce/7eea0c2703631e73a3ad1a945c3b36c5.
> 
> Is this still the smallest combo found?

Yes. With that combo if you place "floor" blocks from the Platforms mod in an upside down L shape, they stop rendering. To break other mods, you will need to add the other 2 mods I mentioned. As long as the stuff from Platforms is broken, the issue only gets worse the more mods you add.

**Comment by TropheusJ** [comment]
something else to consider is that it might be an issue with *models*, which would make more sense really
see if you can get your test mod to print block model count

**Comment by itsdinkd** [comment]
> 

![image](https://user-images.githubusercontent.com/35115304/166168453-de566bab-1d32-4948-a90a-98e284cfaa0e.png)

**Comment by itsdinkd** [comment]
Ah youre using Universal Graves. your issue is because of Polymer. Its a dependency of universal graves. I had this problem on 1.18.1 where blocks wouldn't render, forcing me to use Youre in grave danger. I told Patbox about this but he couldn't find what was wrong exactly.

**Comment by Footage2-Amply-Pounce** [comment]
Unfortunately, I can't test it right now. Since the last time I had C+, I added a bunch of new mods, and now adding C+ prevents me from joining the server with a generic "Invalid player data" error. I will have to figure that one out later and, if it has anything to do with C+, open a separate issue for it. I would test with just the necessary mods, rather than my entire modpack, but I can't do that today.

Meanwhile if someone can test if Universal Graves/Polymer is indeed what's causing it, I'd appreciate it. I highly suspect that's the culprit.

**Comment by Footage2-Amply-Pounce** [comment]
Ok, even worse news for me, it's not adding C+ that corrupts my player data, it's removing UG...

Anyway, I tested it and can confirm that Universal Graves in Combination with C+ causes multiblocks to break when using more than at least 14 mods. If you remove either UG or C+, the issue goes away. Should I keep the issue open on your side? 

I can test things further if needed.

@Patbox would you be able to help?

**Comment by Patbox** [comment]
First try updating standalone polymer version (https://www.curseforge.com/minecraft/mc-mods/polymer/files). Second install this mod https://github.com/Tom-The-Geek/WDMNF/releases/tag/v1.1.0%2Bsnapshots

**Comment by Footage2-Amply-Pounce** [comment]
> First try updating standalone polymer version (https://www.curseforge.com/minecraft/mc-mods/polymer/files). Second install this mod https://github.com/Tom-The-Geek/WDMNF/releases/tag/v1.1.0%2Bsnapshots

Updating Polymer didn't help. I removed its configs and created a new world just to be sure.

Logs with WDMNF: https://gist.github.com/Footage2-Amply-Pounce/c55bc2c2277a2520e400a5c7dcdabf78. Note: this is my entire modpack, so if it's difficult to read logs with so many mods, I can get a log with a small amount. I didn't this time because for whatever reason my game is f-ing with me right now and I couldn't reproduce the issue with the list I provided earlier, so I just loaded my actual modpack for convenience's sake.

**Comment by Patbox** [comment]
Could you try removing signEditor mod?

**Comment by Footage2-Amply-Pounce** [comment]
Just for the record it's [Better Signs and Frames](https://www.curseforge.com/minecraft/mc-mods/better-signs-and-frames). I noticed the error it prints every time I join a while ago, but the only thing that's broken from that mod is that there's no hand animation when opening containers through signs and frames. Removing the mod doesn't affect the issue.

Logs without Better Signs: https://gist.github.com/Footage2-Amply-Pounce/12120fd8326903c3ba61797f66ec3c83

**Comment by Patbox** [comment]
Could you send client log? Ideally with both updated polymer and wdmnf

**Comment by Footage2-Amply-Pounce** [comment]
The gist includes both the client and the server logs, I marked them accordingly. With updated Polymer. Unless you mean to load with *just* Polymer and WDMNF?

**Comment by Patbox** [comment]
Ok, I didn't notice there were 2 log files. I will look into that

**Comment by Patbox** [comment]
Ok, please run `/polymer export-registry` and send generated file here

**Comment by Footage2-Amply-Pounce** [comment]
The file is massive, so I have to attach it as a zip: [polymer-dump-client.zip](https://github.com/ConsistencyPlus/ConsistencyPlus/files/8603550/polymer-dump-client.zip)

**Comment by Patbox** [comment]
Could you check if disabling polymer networking in config/polymer/client.json fixes it?

**Comment by Footage2-Amply-Pounce** [comment]
> I assume if you change it on the client as well, the same thing will happen in SP as it's considered a local server.

Yep, just tested it and setting it to false in server.json instead of client.json made me unable to join any worlds as well, as if I removed the mod. I'm so lost with this to be honest.

**Comment by Footage2-Amply-Pounce** [comment]
I disabled polymer networking, deleted all my config files and tried to join an existing server, this time without Sodium. Instead of invisible chunks, it flat out crashes when generating new chunks: https://gist.github.com/Footage2-Amply-Pounce/0fe2389d64da4e3be3708a5161985c68. The silver-lining is that it indeed fixes the original issue, lol.

I will be reassembling my modpack in parts, because not even a fresh install of it minus UG lets me create worlds, will report if I find anything useful. But for now at least it's confirmed that Polymer's networking is what's causing my original issue.

**Comment by Patbox** [comment]
It only happens if you have C+ and Polymer right?

**Comment by Patbox** [comment]
It's possible issue is somewhere else and polymer just "saved" you from instant crashing (it has some failsaves). It's possible you have (somehow) too many blocks

**Comment by Footage2-Amply-Pounce** [comment]
> It only happens if you have C+ and Polymer right?

Yep, only when both are present and there's at least like 15 mods that add new blocks and multiblocks. For some reason it's inconsistent at which point things start to break (I swear I'm not trying to waste your time), but once you hit it, then after that point the more mods you add, the worse it gets.

> It's possible issue is somewhere else and polymer just "saved" you from instant crashing (it has some failsaves). It's possible you have (somehow) too many blocks

Not out of the question, I'm just shy of 300 jars in my mods folder, so I'm grateful you guys are even trying to help at all instead of shrugging it off as "you have way too many mods" 😄 Hopefully rebuilding the pack will help.

**Comment by itsdinkd** [comment]
Good luck!

**Comment by Patbox** [comment]
Polymer complained about negative blockstate ids. Which honestly not sure why would happen otherwise

**Comment by Footage2-Amply-Pounce** [comment]
> > It's possible issue is somewhere else and polymer just "saved" you from instant crashing (it has some failsaves). It's possible you have (somehow) too many blocks
> 
> Not out of the question, I'm just shy of 300 jars in my mods folder, so I'm grateful you guys are even trying to help at all instead of shrugging it off as "you have way too many mods" 😄 Hopefully rebuilding the pack will help.

Alright, after rebuilding my modpack I found out that [Hyper-Lighting](https://github.com/Exploding-Creeper/Hyper-Lighting) was simply incompatible with [Colormatic](https://www.curseforge.com/minecraft/mc-mods/colormatic). That's what caused invisible chunks with Sodium and crashes without. It tried to render Coloured water lakes. I reported it to them and deleted the crashlog from this thread as it's irrelevant to this issue.

And "Invalid Player Data" was caused by [Roughly Enough Resources](https://www.curseforge.com/minecraft/mc-mods/roughly-enough-resources). I used to have the same issue with RER on 1.17.1 with a large modpack. Someone else already submitted a report to them, it may have something to do with datapacks. I have no idea why it worked with Polymer present.

Other than that my modpack seems to be running fine now without Universal Graves.

I'm leaving this issue open since there's still some sort of incompatibility between C+ and UG. If you still need me to test something, I'll be glad to help. For the time being, I will have to make a decision between UG and C+.

**Comment by Patbox** [comment]
Ok, could you send polymer export-registry from both server and client?

**Comment by Footage2-Amply-Pounce** [comment]
> Ok, could you send polymer export-registry from both server and client?

Server: [polymer-dump-server.zip](https://github.com/ConsistencyPlus/ConsistencyPlus/files/8605573/polymer-dump-server.zip)
Client: [polymer-dump-client.zip](https://github.com/ConsistencyPlus/ConsistencyPlus/files/8605581/polymer-dump-client.zip)

**Comment by Patbox** [comment]
> > Ok, could you send polymer export-registry from both server and client?
> 
> Server: [polymer-dump-server.zip](https://github.com/ConsistencyPlus/ConsistencyPlus/files/8605573/polymer-dump-server.zip) Client: [polymer-dump-client.zip](https://github.com/ConsistencyPlus/ConsistencyPlus/files/8605581/polymer-dump-client.zip)

You have mismatched server and client mods (there is lack of some mods on server). It causes block ids to desync, which will make blocks display as wrong ones.

**Comment by Footage2-Amply-Pounce** [comment]
> You have mismatched server and client mods (there is lack of some mods on server, non-polymer ones). It causes block ids to desync, which will make blocks display as wrong ones.

The only mod I could find that wasn't on the server is Gromov's Paintings, must've forgotten to throw it in. Other than that the server has all the mods that need to be on it, all the other ones are exclusively client-side. There are also some exclusively server-side mods that do nothing on the client. What mods were you referring to, specifically? I could try running all the mods on both sides, but as far as I'm aware as long as mods don't change blockstates there won't be any issues connecting to servers that don't have it. And if you don't have some necessary mods that the server does, it won't even let you in and show a list of missing block ids.

**Comment by Patbox** [comment]
Visualoverhaul is one of mods (it registers new block). Not sure about other

**Comment by Footage2-Amply-Pounce** [comment]
VisualOverhaul is present on both sides, you can see it in my [logs](https://gist.github.com/Footage2-Amply-Pounce/c55bc2c2277a2520e400a5c7dcdabf78) I linked earlier. Also it shouldn't matter [according to the dev](https://www.curseforge.com/minecraft/mc-mods/visual-overhaul?comment=81).

Just loaded up with identical mods to make sure (deleted every single mod from the server and copied over all my client mods) - the issue is still present.

**Comment by Cart3r1234** [comment]
So just to catch up on the new info that's been discovered on this in the last day
- its caused by having C+ and Polymer in a pack at the same time
- it might have something to do with blockstate count breaking

Have any other large block mods been tested with polymer to see if the same problem occurs, or is it _just_ Consistency+ somehow

**Comment by Footage2-Amply-Pounce** [comment]
>  its caused by having C+ and Polymer in a pack at the same time

Yes

>  it might have something to do with blockstate count breaking

I don't feel like I know enough about modding to say for sure, but it might.

> Have any other large block mods been tested with polymer to see if the same problem occurs, or is it _just_ Consistency+ somehow

Yes, out of a couple hundreds of mods that I use, only C+ and [Jello](https://www.curseforge.com/minecraft/mc-mods/jello) cause this, when used together with Polymer. I'm not sure if Jello causes it because of the blockstate count (or whatever issue it ends up being) or because they use something from C+ that's causing it, as they have C+ listed as an optional dependency. I'm not qualified enough to look through their source code to find that out. And I don't know of any other mods that add this many blocks. Chipped kinda does, but there're no issues with it.

**Comment by Dragon-Seeker** [comment]
>  ...as they have C+ listed as an optional dependency. 
It was a optional dependency since I had coded up some interaction with the mod but had disabled it during the updating to 1.18.2 since I was waiting on them and I kinda redid the whole system.

**Comment by Dragon-Seeker** [comment]
Frankly this seems like we have reached a limitation of minecraft somehow

**Comment by Footage2-Amply-Pounce** [comment]
> It was a optional dependency since I had coded up some interaction with the mod but had disabled it during the updating to 1.18.2 since I was waiting on them and I kinda redid the whole system.

Ah, got it. So it must be that whenever some amount of blocks/blockstates/whatever crosses a certain number, Polymer "breaks", subsequently breaking multiblocks from all sorts of mods. If that's not it, I have not the slightest clue what else could it be.

> Frankly this seems like we have reached a limitation of minecraft somehow

Wouldn't that mean that in that case I would be able to reproduce this issue without Polymer? And in single-player as well? Polymer doesn't add anything new as far as I'm aware, it just handles networking stuff.

**Comment by Dragon-Seeker** [comment]
True. my bad but It seems then a problem with the amount of blocks in which creating to many block states for polymer to handle?

**Comment by Dragon-Seeker** [comment]
Is it possible to send a full list of the incompatible mods?

**Comment by Footage2-Amply-Pounce** [comment]
> True. my bad but It seems then a problem with the amount of blocks in which creating to many block states for polymer to handle? But I don't understand why such matters as Polymer shouldn't care about those blocks right?

Your guess would be much better than any of mine 😄 As I told Patbox, I'm basically open to more testing, but for now I will just have to be aware that really large modpacks might misbehave with Polymer.

> Is it possible to send a full list of the incompatible mods?

Basically C+/Jello (or both if you want to be sure) + UniversalGraves/Polymer and this list on top: https://gist.github.com/Footage2-Amply-Pounce/7eea0c2703631e73a3ad1a945c3b36c5. Most of these are really mostly needed because they are the aforementioned multiblocks mods and without those mods there's nothing to test. I used to be able to reproduce the issue with a couple less mods just a few days ago, but today when I was testing with Patbox, for some reason 14 mods wasn't enough 🤷 Just like initially, Waystones used to break with a minimal amount of mods, yet after I ran the test mod from Siuolthepic, I couldn't break Waystones with just these mods anymore, but still could with the rest of the modpack. So if you can't reproduce the issue with these mods, try adding more, although I think C+ and Jello in combination with the list I linked should definitely be enough.

**Comment by Patbox** [comment]
Generally Polymer shouldn't change raw ids (outside of moving polymer blockstates at end of state list), because it only needs to replace "PolymerObjects" with vanilla (client) compatible representation. BlockState hack is required for mod compatiblity, but only thing it really does is moving adding PolymerBlock's state in time to last possible moment. It should preserve order of other elements. It really sounds like BlockState desync caused by missing regular mods (There are 729748 non-polymer BlockStates on server and 729751 on client). One is `suggestionproviderfix` and other `visualoverhaul`. So first try to fix these

**Comment by Footage2-Amply-Pounce** [comment]
> One is `suggestionproviderfix` and other `visualoverhaul`. So first try to fix these

I went ahead and reproduced the issue with just the mods I linked in my previous message, without these 2. Here are the dumps:

[polymer-dump-server.zip](https://github.com/ConsistencyPlus/ConsistencyPlus/files/8610038/polymer-dump-server.zip)
[polymer-dump-client.zip](https://github.com/ConsistencyPlus/ConsistencyPlus/files/8610040/polymer-dump-client.zip)

Example of how its broken with these mods:

There's 2 platforms on top of each other, and 2 glass panes on top of each other. When placed immediately they became invisible, then I right clicked the ground between them and it refreshed the bottom blocks
![2022-05-03_13 07 40](https://user-images.githubusercontent.com/86702831/166443207-7aa39094-4541-413d-99cb-41e3af8d430c.png)

**Comment by Patbox** [comment]
Does disabling polymer networking (without changing anything else!) fixes that?

**Comment by Footage2-Amply-Pounce** [comment]
Just tested it (set it to false in client.json and server.json on the client and in server.json on the server) and nope, it doesn't fix it. So it's not the networking part that's messing stuff up? I'm lost...

**Comment by Patbox** [comment]
[polymer-test.zip](https://github.com/ConsistencyPlus/ConsistencyPlus/files/8610820/polymer-test.zip)
Ok. Install this version of polymer (both server and client) and check if anything changed. Additionally, please send screenshot of F3 screen and output of `/polymer target-block` command while targetting this block

**Comment by Footage2-Amply-Pounce** [comment]
I observed no changes, with and without networking set to false.

<details>
  <summary>Screenshots</summary>
  
Initially placed, 1 block:
![2022-05-03_15 52 25](https://user-images.githubusercontent.com/86702831/166467369-89602fc1-4c17-42fb-b78e-21b555978e8b.png)

Same, but using `/polymer target-block`:
![2022-05-03_15 54 43](https://user-images.githubusercontent.com/86702831/166467807-dac45f9c-5709-4f26-8252-d779c4212293.png)

Another block placed directly on top of it:
![2022-05-03_15 52 28](https://user-images.githubusercontent.com/86702831/166467525-18ada86c-d207-4417-992b-eb9554659d7c.png)

Same, but using `/polymer target-block`:
![2022-05-03_15 54 53](https://user-images.githubusercontent.com/86702831/166467860-2d865c86-940e-49af-85e0-8fbc7de33d83.png)

Right-clicked the ground next to the bottom block with an empty hand:
![2022-05-03_15 52 33](https://user-images.githubusercontent.com/86702831/166467601-79aedc10-6f87-4f27-ae30-af99922153ef.png)

Same, but using `/polymer target-block`:
![2022-05-03_15 55 06](https://user-images.githubusercontent.com/86702831/166467898-84454cb4-e8a0-4d8c-a829-fe01ad05866f.png)

</details>

If I'm reading this correctly, seems like Polymer/the server is aware there's a block and which block it is, but not the client.

**Comment by Patbox** [comment]
So after all it is blockstate desync. I will wrote more stuff to validate where it breaks

**Comment by Patbox** [comment]
[polymer-test.zip](https://github.com/ConsistencyPlus/ConsistencyPlus/files/8611233/polymer-test.zip)
Could you try updating polymer on client and server again to test if it fixes it?

**Comment by Footage2-Amply-Pounce** [comment]
> [polymer-test.zip](https://github.com/ConsistencyPlus/ConsistencyPlus/files/8611233/polymer-test.zip) Could you try updating polymer on client and server again to test if it fixes it?

Sadly, still not. Tried with both networking sync off and on.

**Comment by Patbox** [comment]
[poltest.zip](https://github.com/ConsistencyPlus/ConsistencyPlus/files/8611367/poltest.zip)
Could you try this one? Keep in mind to install it on both sides!

**Comment by Footage2-Amply-Pounce** [comment]
Nope, still not fixed. And I'm installing it on both sides, of course.

**Comment by Patbox** [comment]
[poltest.zip](https://github.com/ConsistencyPlus/ConsistencyPlus/files/8611816/poltest.zip)

Yet another test version. Install on both client and server. Just remember to have enabled networking!

While connected to the server, just type `/polymer dev validate_states`. It should print bunch of messages into chat! After that please send me client logs!

**Comment by Footage2-Amply-Pounce** [comment]
> While connected to the server, just type `/polymer dev validate_states`. It should print bunch of messages into chat! After that please send me client logs!

The issue still remains, logs: https://gist.github.com/Footage2-Amply-Pounce/dfe8ded54ad19b6d7c33b48c492c8302

**Comment by Patbox** [comment]
[poltest.zip](https://github.com/ConsistencyPlus/ConsistencyPlus/files/8611939/poltest.zip)
Could you retry to generate with this one? Previous had small bug. Do the same thing as last time

**Comment by Footage2-Amply-Pounce** [comment]
Updated to the same gist: https://gist.github.com/Footage2-Amply-Pounce/dfe8ded54ad19b6d7c33b48c492c8302

This time it didn't print anything in the chat, though.

**Comment by Footage2-Amply-Pounce** [comment]
The issue is still there. 

Same order of screenshots as last time: F3 - polymer command etc.

<details>
  <summary>Screenshots</summary>
  
![2022-05-03_18 36 49](https://user-images.githubusercontent.com/86702831/166499988-b9e1ed27-0369-474c-ac18-c02d0fe484f0.png)
![2022-05-03_18 36 56](https://user-images.githubusercontent.com/86702831/166499999-f3e0a011-0a4e-4b5b-a02d-74f343ad572d.png)
![2022-05-03_18 37 03](https://user-images.githubusercontent.com/86702831/166500006-10e20acc-0d0a-42db-b708-0ce390bc33ab.png)
![2022-05-03_18 37 10](https://user-images.githubusercontent.com/86702831/166500017-d89ceff2-edb7-4946-bc25-201b16dd8394.png)
![2022-05-03_18 37 18](https://user-images.githubusercontent.com/86702831/166500022-26de4aa8-b67d-4ba0-adf3-f9b3beb27989.png)
![2022-05-03_18 37 25](https://user-images.githubusercontent.com/86702831/166500032-30239939-f46a-4c44-8feb-8e44917b5464.png)


</details>

**Comment by Patbox** [comment]
Ok ids are correct this time! Could you check if these are blockentities with `/data get block...` command?

**Comment by Footage2-Amply-Pounce** [comment]
Like so?

![2022-05-03_18 54 03](https://user-images.githubusercontent.com/86702831/166501925-563649af-fb07-4edf-a38e-3a69c2437bfb.png)

**Comment by Footage2-Amply-Pounce** [comment]
Note: it shows correctly here, because I left and rejoined the server. Such method always made it render properly until updated again, e.g. by breaking a block near it.

**Comment by Patbox** [comment]
So it's block entity issue... Could you check if other previously effected blocks still have issues?

**Comment by Footage2-Amply-Pounce** [comment]
Yes, they are still affected. Not all of them are block entities, though.

<details>
  <summary>Screenshots</summary>

![2022-05-03_19 00 20](https://user-images.githubusercontent.com/86702831/166502991-ac225a37-5434-4ae8-8603-54ae4d74fafb.png)

![2022-05-03_19 00 27](https://user-images.githubusercontent.com/86702831/166503001-fcf3f1ab-d3e7-4700-b233-3bcd1383a211.png)

![2022-05-03_19 00 35](https://user-images.githubusercontent.com/86702831/166503010-e711568d-6319-481a-bdbf-bda5506b6776.png)


</details>

**Comment by Patbox** [comment]
Does the pane one fix itself if you click top of block?

**Comment by Patbox** [comment]
[poltest.zip](https://github.com/ConsistencyPlus/ConsistencyPlus/files/8612333/poltest.zip)

I have finally good feeling about this... Could you check if this fixes it?

**Comment by Footage2-Amply-Pounce** [comment]
Well, something has certainly changed, but not fixed yet.

When I place the glass panes on top of each other, same issue - they turn invisible, I right-click the ground, bottom one shows, but not the top one.
When I place platforms on top of each other, now the first one will turn invisible, but after clicking the ground block near it, it renders properly, without breaking, but a platform block placed on top, still has the old behavior.
When I place a multibed down, half of it turns invisible until I right-click the ground, but the texture is rendered correctly; and when you place another bed next to it so they connect, halves of them stop rendering in a cross pattern.

![2022-05-03_19 22 37](https://user-images.githubusercontent.com/86702831/166506041-19179679-6969-4738-b743-e701030490e2.png)

**Comment by Patbox** [comment]
Ok, could you check if disabling `sendBlocksAroundClicked` in `config/polymer/server.json` (on server ofc) fixes that?

**Comment by Footage2-Amply-Pounce** [comment]
Nope, it only broke the textures again.

![2022-05-03_19 31 40](https://user-images.githubusercontent.com/86702831/166507388-abe3c7ac-ff33-4403-bf26-a3cbc4853daf.png)

**Comment by Patbox** [comment]
Do you still have logs with about invalid blockstates with negative ids?

**Comment by Footage2-Amply-Pounce** [comment]
These ones? https://gist.github.com/Footage2-Amply-Pounce/dfe8ded54ad19b6d7c33b48c492c8302

**Comment by Patbox** [comment]
Yes... Hmm

**Comment by Patbox** [comment]
Another question. Do you get these errors packets after placing these blocks or on join?

**Comment by Footage2-Amply-Pounce** [comment]
After placing/breaking the blocks themselves or blocks around them. If you just join and don't do anything, just walk around, everything looks normal.

**Comment by Patbox** [comment]
[poltest.zip](https://github.com/ConsistencyPlus/ConsistencyPlus/files/8612840/poltest.zip)
Another test! I don't have too hight hopes, but it's worth a shot

**Comment by Footage2-Amply-Pounce** [comment]
Something went very wrong this time. Seems like the client now also doesn't receive proper blockstates causing this:

![2022-05-03_20 41 15](https://user-images.githubusercontent.com/86702831/166521129-269d40ce-862d-4cc0-8a58-5e258c41cb28.png)

I've only seen this happen when a client joins a server that has a mod, that changes blockstates, or vice versa. So I made absolutely sure I didn't miss anything and uploaded the mod to both sides.

**Comment by Patbox** [comment]
Removing polymer and universal graves (check /polymer command on client and server) fixes these issues right?

Could you check if it happens without polymer, with polymer only on server (with and without UG) and with polymer only on client?
[poltest.zip](https://github.com/ConsistencyPlus/ConsistencyPlus/files/8612898/poltest.zip)

**Comment by Footage2-Amply-Pounce** [comment]
> Removing polymer and universal graves (check /polymer command on client and server) fixes these issues right?

Correct.

> Could you check if it happens without polymer

Since UG bundles Polymer, I have to remove it as well. In that case everything works as intended.

> with polymer only on server (with and without UG)

Polymer on the server, not on the client, no UG - no issues at all!
Polymer on the server, not on the client, UG is present - no issues at all!

> with polymer only on client?

Polymer not on the server, present on the client - good old disappearing acts and missing textures.

**Comment by Patbox** [comment]
Disabling networking didn't fix that right?

**Comment by Footage2-Amply-Pounce** [comment]
Nope, disabling networking had no effect.

**Comment by Patbox** [comment]
Okay, does this version also have that issue?
[poltest.zip](https://github.com/ConsistencyPlus/ConsistencyPlus/files/8613107/poltest.zip)

**Comment by Footage2-Amply-Pounce** [comment]
> Okay, does this version also have that issue? [poltest.zip](https://github.com/ConsistencyPlus/ConsistencyPlus/files/8613107/poltest.zip)

No. Seems like you've fixed it! I'm going to launch my full modpack just to be sure and if it still doesn't break, I guess it's fixed.

What was the problem? Specifically, I'm curious why would it only happen when C+ or Jello is present, but otherwise despite my modpack still being massive, it was totally fine?

**Comment by Footage2-Amply-Pounce** [comment]
Oh, alright, but just so you know, even my full modpack seems to be working well with that mixin disabled.

> Could you test this?
> [poltest.zip](https://github.com/ConsistencyPlus/ConsistencyPlus/files/8613728/poltest.zip)

No issues :)

![2022-05-03_21 41 57](https://user-images.githubusercontent.com/86702831/166553582-816fd624-5375-4cd2-892f-6c40e2d8042c.png)

**Comment by Patbox** [comment]
Well, I guess it is conflict with one of these mods. It does something hacky I'm questioning even why

**Comment by Patbox** [comment]
Could you use this and send me logs after placing some of effected blocks?
[poltest.zip](https://github.com/ConsistencyPlus/ConsistencyPlus/files/8613845/poltest.zip)

**Comment by Footage2-Amply-Pounce** [comment]
> Could you use this and send me logs after placing some of effected blocks? Want to check something (only client needs to update) [poltest.zip](https://github.com/ConsistencyPlus/ConsistencyPlus/files/8613845/poltest.zip)

I see a bunch of errors in the log, but seemingly everything worked as intended: https://gist.github.com/Footage2-Amply-Pounce/6ffa1ca8b06f567ab9977a2c8bf3bb40

**Comment by Patbox** [comment]
Okay, does this one fix the errors?
[poltest.zip](https://github.com/ConsistencyPlus/ConsistencyPlus/files/8613952/poltest.zip)

**Comment by Footage2-Amply-Pounce** [comment]
> Okay, does this one fix the errors? [poltest.zip](https://github.com/ConsistencyPlus/ConsistencyPlus/files/8613952/poltest.zip)

No, I updated the gist with the new logs, but it's all the exact same errors I think: https://gist.github.com/Footage2-Amply-Pounce/6ffa1ca8b06f567ab9977a2c8bf3bb40

**Comment by Patbox** [comment]
[poltest.zip](https://github.com/ConsistencyPlus/ConsistencyPlus/files/8614085/poltest.zip)
Remember to update in on both server and client! Place some of effected blocks (mostly on client, thats for debugging)

**Comment by Footage2-Amply-Pounce** [comment]
The server kicked me out when I broke one of the multiblocks, didn't crash though: https://gist.github.com/Footage2-Amply-Pounce/3217082f77e8d625b1aab4f996d87b56 (both client and server logs are included).

**Comment by Patbox** [comment]
The plot thickens... this error means it's server writing invalid ids. I don't have time now, so I will continue working on it tomorrow

**Comment by Footage2-Amply-Pounce** [comment]
Yep, no problem. I will be updating my modpack, but will keep this copy for further testing.

**Comment by Patbox** [comment]
Experimented with it more and... It's not polymer bug. Or well, it is? Vanilla seems to create these negative ids by itself which resolve to null. While MC normally ignores them, polymer replaced it with air. This then removes blocks from client world causing initial issue. I guess I will make polymer emulate vanilla behaviour, but keep in minds it might become issue later again...

**Comment by Patbox** [comment]
Next polymer update (0.2.0-pre.2+1.18.2) should (un)fix polymer so it won't replace invalid values. It should be available on CF and modrinth in a moment. You can add it to your pack for now, before there is a proper fix

**Comment by Cart3r1234** [comment]
Closing this since the vanilla issue that caused this has been fixed
